pkg/resources/execute.go

```diff
@@ -0,0 +1,172 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/resources"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var executeSchema = map[string]*schema.Schema{
+	"execute": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "SQL statement to execute. Forces recreation of resource on change.",
```

Comment:
Nit: we can also add that the statement should be idempotent.

Reply:
It does not need to, though. That's why there is also revert function: when any change happens to execute, it is first reverted.